### PR TITLE
feat(logs): import a LogGroup from its name

### PR DESCRIPTION
**Pull request by nija-at**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-logs/lib/log-group.ts [hunk 1]
```diff
@@ -316,6 +316,23 @@ export class LogGroup extends LogGroupBase {
     return new Import(scope, id);
   }
 
+  /**
+   * Import an existing LogGroup given its name
+   */
+  public static fromLogGroupName(scope: Construct, id: string, logGroupName: string): ILogGroup {
+    class Import extends LogGroupBase {
```
Comment (eladb) [inline packages/@aws-cdk/aws-logs/lib/log-group.ts]: can we reuse `fromLogGroupArn`?

Comment (nija-at) [reply packages/@aws-cdk/aws-logs/lib/log-group.ts]: Why is that? Can you elaborate?

Comment (eladb) [reply packages/@aws-cdk/aws-logs/lib/log-group.ts]: Just less duplication (not that there is a lot, but still).

Comment (eladb) [reply packages/@aws-cdk/aws-logs/lib/log-group.ts]:
```ts
public static fromLogGroupName(scope: Construct, id: string, logGroupName: string): ILogGroup {
  return this.fromLogGroupArn(scope, id, Stack.of(scope).formatArn({
    service: 'logs', resource: 'log-group', sep: ':', resourceName: logGroupName
  });
}
```

Comment (nija-at) [reply packages/@aws-cdk/aws-logs/lib/log-group.ts]: Not a fan of this actually. Doing this makes the ARN a `Fn::Join` token and the loggroup name becomes a further token using more CF intrinsics. While all of this is technically valid, it seems to make things a bit complex.

I'd rather leave this as is.

**Pull request merged**